sklearn/utils/extmath.py

```diff
@@ -599,8 +599,10 @@ def log_logistic(X, out=None):
     http://fa.bianp.net/blog/2013/numerical-optimizers-for-logistic-regression/
     """
     is_1d = X.ndim == 1
+    X = np.atleast_2d(X)
```

Comment:
This is weird, to call `atleast_2d` before `check_array`. Why not pass `ensure_2d` to `check_array` as X is expected to be 2 dimensional based on the docstring?